testutils/src/main/java/com/microsoft/identity/internal/testutils/labutils/LabGuestAccountHelper.java

```diff
@@ -33,9 +33,14 @@ public static LabGuest loadGuestAccountFromLab(final LabUserQuery query) {
 
         return new LabGuest(
                 userInfo.getHomeUPN(),
-                userInfo.getLabName(),
+                userInfo.getHomeDomain(),
                 userInfo.getHomeTenantID(),
                 guestLabTenants
         );
     }
+
+    public static String getPasswordForGuestUser(final LabGuest guestUser) {
+        final String labName = guestUser.getHomeDomain().split("\\.")[0];
+        return LabHelper.getSecret(labName);
+    }
```

Comment:
Do this existing mechanism for getting password not work for your needs?

Reply:
No they get password for the msidlab4 (where the user is guest), we need the password for the home lab

Reply:
Are you saying that the existing guest account helper code fetches password for the wrong lab? How are they existing guest account tests working then?

Reply:
I did not say that. 
What I am saying is that existing code fetches the password for the lab, which gets returned in the labConfig part of the config endpoint response. In this case that labconfig is for msidlab4 which is serving the account as it exists as a guest in that lab.
If things are not clear, You can attach fiddler or debug through existing test case to see what I am saying

Reply:
So this seems a change in behavior by the lab api...for regular guest users (in the same cloud), the lab info returned was lab of the home cloud / tenant, however, it seem for cross cloud guests, the lab info returned is now of the guest cloud / tenant

Reply:
we should probably ask the lab team to fix it

Reply:
Resolved offline